src/main/java/jenkins/plugins/git/GitSCMFileSystem.java

```diff
@@ -283,15 +284,66 @@ public boolean supportsDescriptor(SCMSourceDescriptor descriptor) {
             return AbstractGitSCMSource.class.isAssignableFrom(descriptor.clazz);
         }
 
+        public static class HeadNameResult
```

Comment:
```suggestion
        static class HeadNameResult
```
Not used outside this class and the test, right?

(In which case you could also simplify a bit by dropping `private` from the fields and removing the getters—this is just an internal struct, no need for accessors.)